x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/EsqlMediaTypeParser.java

```diff
@@ -43,6 +43,23 @@ public static MediaType getResponseMediaType(RestRequest request, EsqlQueryReque
         return validateColumnarRequest(esqlRequest.columnar(), mediaType, request);
     }
 
+    /*
+     * If requests have no media type in params nor headers, default to JSON
+     */
+    public static MediaType getResponseMediaType(RestRequest request) {
+        MediaType mediaType;
+        if (request.hasParam(URL_PARAM_FORMAT)) {
+            mediaType = mediaTypeFromParams(request);
+        } else {
+            ParsedMediaType acceptType = request.getParsedAccept();
+            mediaType = acceptType != null ? acceptType.toMediaType(MEDIA_TYPE_REGISTRY) : request.getXContentType();
+        }
+        if (mediaType == null) {
```

Comment:
This should not happen. Check the comments on top of the other `getResponseMediaType()` method.